### PR TITLE
change to use IP address instead of hostname for nb and sb db creation

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
On azure the hostnames do not get resolved correctly, change db creation
to use the ip address instead

### CODE REVIEW

**Comment by dcbw** [comment]
> does this work with HA (raft)?

@pecameron yeah, this is because of RAFT. For whatever reason, OVS isn't able to resolve the machine's hostname to an IP on Azure. Maybe something to do with the lack of dots in the name, or who knows why (since `host -T <name>` and `getent ahostsv4 <name>` both work).

**Comment by dcbw** [comment]
@JacobTanenbaum legit e2e failure in nbdb pods:

```
+ [[ -f /env/_master ]]
+ OVN_NODES_ARRAY=(ip-10-0-130-61.ec2.internal ip-10-0-136-10.ec2.internal ip-10-0-144-47.ec2.internal)
/bin/bash: -c: line 8: unexpected EOF while looking for matching `)'
```

**Comment by squeed** [comment]
I'm confused - you're saying the node names cannot be resolved, but you resolve them with `getent`?

Let's just switch to generating an array of IP addresses by pulling them from the Node object in the operator.

**Comment by dcbw** [comment]
> I'm confused - you're saying the node names cannot be resolved, but you resolve them with `getent`?
> 
> Let's just switch to generating an array of IP addresses by pulling them from the Node object in the operator.

@squeed yes, in Azure for whatever reason the node names cannot be resolved by OVS. getent works though.

**Comment by squeed** [comment]
Wild.

File: bindata/network/ovn-kubernetes/ovnkube-master.yaml [hunk 1]
```diff
@@ -111,10 +111,13 @@ spec:
           fi
           
           OVN_NODES_ARRAY=({{.OVN_NODES}})
-          if [[ "$K8S_NODE" == "${OVN_NODES_ARRAY[0]}" ]]; then
+          MASTER_NODE=$(getent ahostsv4 "${OVN_NODES_ARRAY[0]}" | grep RAW | awk '{print $1}'
+          LOCALHOST=$(getent ahostsv4 "${K8S_NODE}" | grep RAW | awk '{print $1}')
```
Comment (danwinship) [inline bindata/network/ovn-kubernetes/ovnkube-master.yaml]: `ahosts`, not `ahostsv4`. At this point everyone needs to be keeping future IPv6 compatibility in mind with all new code.

Comment (JacobTanenbaum) [reply bindata/network/ovn-kubernetes/ovnkube-master.yaml]: done

**Comment by dcbw** [comment]
/test e2e-aws-ovn-kubernetes

```
time="2019-11-13T15:53:22Z" level=debug msg="module.dns.aws_route53_record.etcd_cluster: Creation complete after 56s [id=Z08278162J4YYMJY8PNI9__etcd-server-ssl._tcp_SRV]"
time="2019-11-13T15:53:22Z" level=error
time="2019-11-13T15:53:22Z" level=error msg="Error: NoSuchBucket: The specified bucket does not exist"
time="2019-11-13T15:53:22Z" level=error msg="\tstatus code: 404, request id: 7AC51675A008BB7F, host id: wBF7sI1/FUpciSIdRtGFVZ6X+UWaPAeWHVDGsAhOPBabW/uenDBR7LHCULS9FL3ltaohbCsjMOw="
time="2019-11-13T15:53:22Z" level=error
time="2019-11-13T15:53:22Z" level=error msg="  on ../tmp/openshift-install-134534759/bootstrap/main.tf line 5, in resource \"aws_s3_bucket\" \"ignition\":"
time="2019-11-13T15:53:22Z" level=error msg="   5: resource \"aws_s3_bucket\" \"ignition\" {"
time="2019-11-13T15:53:22Z" level=error
time="2019-11-13T15:53:22Z" level=error
time="2019-11-13T15:53:22Z" level=fatal msg="failed to fetch Cluster: failed to generate asset \"Cluster\": failed to create cluster: failed to apply using Terraform"
```

**Comment by dcbw** [comment]
This is blocking on https://github.com/openshift/origin/pull/24147 like most other ovn-kube PRs right now.

**Pull request merged**